astropy/cosmology/funcs/comparison.py

```diff
@@ -156,9 +144,9 @@ def _parse_formats(*cosmos: object, format: _FormatsT) -> ndarray:
     # Have to deal with things that do not broadcast well.
     # astropy.row cannot be used in an array, even if dtype=object
     # and will raise a segfault when used in a ufunc.
-    towrap = (isinstance(cosmo, _CosmologyWrapper._cantbroadcast) for cosmo in cosmos)
     wcosmos = [
-        c if not wrap else _CosmologyWrapper(c) for c, wrap in zip(cosmos, towrap)
+        c if not isinstance(c, _CANT_BROADCAST) else _CosmologyWrapper(c)
```

Comment:
If does not matter much, but the double negation made me wonder about alternatives:
1. Simplest: instead of a tuple of classes, have a function `can_broadcast`, so this becomes `c if can_broadcast(c) else _CosmologyWrapper(c)`.
2. Let `_CosmologyWrapper` have a `__new__` that just returns `c` if it is fine -- the class would then be called something like `_WrapIfNeeded` or so. Obviously, this could also be a function `_wrap_if_cannot_broadcast`.

But really, what you have is fine!

Reply:
I like 1. Unfortunately typing doesn't have a `Not` type to do an exclusion.

```python
def can_broadcast(obj: Any) -> TypeGuard[Not[Table, ...]]
```

Reply:
I guess it would be something like `TypeGuard[BroadCastable]` - but not so important for private routine...